### ISSUE TITLE
Warn if a class with const constructor can actually never be const constructed

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#3367]
My understanding from https://github.com/flutter/flutter/issues/102460#issuecomment-1109424737 and https://github.com/dart-lang/language/issues/1296 is that the following class can actually never be const constructed.

```dart
@immutable
class Foo {
  const Foo(this.list) : assert(list.length > 2);
  final List<int> list;
}
```

To make this clear, the `const` should be removed from the constructor. Would it be possible to have a lint for that reminds people to not mark it as const?

**Comment by eernstg** [comment]
dart-lang/language#1296 would actually make `list.length` in the assertion a compile-time error, so this issue could be considered to be a request for a lint in the case where #1296 is rejected.

Also note that dart-lang/language#2219 would introduce support for that expression, so it would just work.
